### PR TITLE
enh: snap-in which uses keyrings to call external APIs

### PR DESCRIPTION
**Pull request by altafDevRev**
## Summary

Adding the example snap-in that provides a command to create an issue in GitHub from an issue in DevRev. The command can be used as: 
```
/gh_issue OrgName RepoName
```

- [x] Added Snap-In
- [x] Update Readme file

For chapter - 5
## Work-item
https://app.devrev.ai/devrev/works/ISS-70830

### CODE REVIEW

File: 9-external-action/code/src/functions/command_handler/index.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  const paramList = paramString.split(' ');
+  if (paramList.length !== 2) {
+    throw new Error('Invalid Parameters');
+  }
+  const [orgName, repoName] = paramList;
+  return [orgName, repoName];
+};
+
+// Function to verify if the orgName is valid
+const verifyOrgName = async (orgName: string, octokit: Octokit) => {
```
Comment (kkviks) [inline 9-external-action/code/src/functions/command_handler/index.ts:37]: Can add type hint for return type.

File: 9-external-action/code/src/functions/command_handler/index.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      org: orgName,
+    });
+  } catch (error) {
+    console.error(error);
+    throw new Error('Invalid Organisation Name');
+  }
+};
+
+// Function to verify if the repoName is valid
+const verifyRepoName = async (orgName: string, repoName: string, octokit: Octokit) => {
```
Comment (kkviks) [inline 9-external-action/code/src/functions/command_handler/index.ts:52]: ditto

File: 9-external-action/code/src/functions/command_handler/index.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+const createGitHubIssue = async (orgName: string, repoName: string, issueDetails: any, octokit: Octokit) => {
+  try {
+    await octokit.request('POST /repos/{owner}/{repo}/issues', {
+      body: issueDetails.description,
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28',
+      },
+      owner: orgName,
+      repo: repoName,
+      title: '[' + issueDetails.issueDisplayName + '] ' + issueDetails.title,
```
Comment (kkviks) [inline 9-external-action/code/src/functions/command_handler/index.ts:77]: optionally:
```suggestion
      title: `[${issueDetails.issueDisplayName}] ${issueDetails.title}`,
```

**Review by kkviks — COMMENTED** [review]
Liked the comprehensive documentation and the updates.

I've provided some suggestions for further improvements. Please take a look when you have a moment. Thanks for the changes!

File: 9-external-action/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+## DevRev Snaps TypeScript Template
+
+This repository contains a template for the functions that can be deployed as
+part of Snap-Ins.
+
+For more reference on Snap-Ins, please refer to the [documentation](https://github.com/devrev/snap-in-docs)
```
[the remark this answers is not in the document]
Comment (altafDevRev) [reply 9-external-action/README.md:6]: Yeah, hadn't updated the readme, will change it according to the snap-in.

**Review by shashankcube — APPROVED** [review]

**Review by coder336 — APPROVED** [review]

**Pull request merged**